src/me/mrCookieSlime/Slimefun/Objects/tasks/ArmorTask.java

```diff
@@ -0,0 +1,87 @@
+package me.mrCookieSlime.Slimefun.Objects.tasks;
+
+import org.bukkit.Bukkit;
+import org.bukkit.entity.Player;
+import org.bukkit.inventory.ItemStack;
+import org.bukkit.potion.PotionEffect;
+import org.bukkit.potion.PotionEffectType;
+
+import me.mrCookieSlime.Slimefun.SlimefunPlugin;
+import me.mrCookieSlime.Slimefun.Lists.SlimefunItems;
+import me.mrCookieSlime.Slimefun.Objects.SlimefunItem.SlimefunArmorPiece;
+import me.mrCookieSlime.Slimefun.Objects.SlimefunItem.SlimefunItem;
+import me.mrCookieSlime.Slimefun.Setup.SlimefunManager;
+import me.mrCookieSlime.Slimefun.api.HashedArmorpiece;
+import me.mrCookieSlime.Slimefun.api.PlayerProfile;
+import me.mrCookieSlime.Slimefun.api.Slimefun;
+import me.mrCookieSlime.Slimefun.api.energy.ItemEnergy;
+import me.mrCookieSlime.Slimefun.utils.Utilities;
+
+public class ArmorTask implements Runnable {
+	
+	private final Utilities utilities = SlimefunPlugin.getUtilities();
+
+	@Override
+	public void run() {
+		for (Player p: Bukkit.getOnlinePlayers()) {
+			if (!p.isValid() || p.isDead()) {
+				continue;
+			}
+			
+			ItemStack[] armor = p.getInventory().getArmorContents();
+			HashedArmorpiece[] cachedArmor = PlayerProfile.get(p).getArmor();
+			
+			for (int slot = 0; slot < 4; slot++) {
+				ItemStack item = armor[slot];
```

Comment:
we should null check and continue

Reply:
It already does a nullcheck in .hasDiverged()
I think it should remain that way, since we otherwise would never clean up the hashed armorpieces if we keep continuing;
Would be a small waste of memory to be honest.